bundles/org.openhab.binding.myuplink/src/main/java/org/openhab/binding/myuplink/internal/handler/MyUplinkGenericDeviceHandler.java

```
@@ -131,10 +133,13 @@ private void updatePropertiesAndOnlineStatus(CommunicationStatus status, JsonObj
     @Override
     public MyUplinkCommand buildMyUplinkCommand(Command command, Channel channel) {
         var deviceId = getConfig().get(MyUplinkBindingConstants.THING_CONFIG_ID).toString();
+        var systemId = getConfig().get(THING_CONFIG_SYSTEM_ID).toString();
```

What will happen, if systemId is not set? Is it ```null```? This would result in a NPE here. If it is just an empty string what will happen later on? The API call will not be valid. I would suggest to add a check before usingthis optional systemId and skip everything which depends on it.

---

That's a good point, will add checks for that

---

A bit tricky to add that check here, since the caller expects a MyUplinkCommand to be returned. It would then be better to add the check in the calling method, but that implementation is in the MyUplinkThingHandler interface, which doesn't have access to the getConfig() method. Would it be ok to move the handleCommand implementation to this class instead and just have a minimal default method in the interface?

---

Or do you think it's better to throw an UnsupportedOperationException from here, since that is already in the default implementation of buildMyUplinkCommand?

---

Went with the second approach, and also added a catch for the exception to prevent throwing it outside the binding.